src/Boxer/Platform/Win32/Win32Application.cpp

```diff
@@ -1,167 +1,352 @@
+		ReleaseDC(dummy_window, dummy_dc);
+		DestroyWindow(dummy_window);
+	}
+
+	Application::Application(String title, U32 width, U32 height) : _Title(title), _Width(width), _Height(height) {
+		Win32Handle* win32 = new Win32Handle();
+		_Handle = win32;
+
+		win32->hInstance = GetModuleHandleW(0);
+		
+		// Register the window class.
+		WNDCLASSEXW wc = {};
+		wc.cbSize = sizeof(wc);
+		wc.style = CS_HREDRAW | CS_VREDRAW | CS_OWNDC;
+		wc.lpfnWndProc = (WNDPROC)WndProc;
+		wc.hInstance = win32->hInstance;
+		wc.lpszClassName = CLASS_NAME;
+		wc.hCursor = NULL;
+
+		if (RegisterClassExW(&wc) == 0) {
+			// TODO(NeGate): We need a logging system
+			ASSERT(false, "Win32: Failed to register window class.");
+		}
+
+		const DWORD style = WS_OVERLAPPEDWINDOW;
+		const DWORD exStyle = WS_EX_APPWINDOW;
+
+		RECT rect;
+		GetClientRect(GetDesktopWindow(), &rect);
+		rect.left = (rect.right / 2) - ((_Width + 16) / 2);
+		rect.top = (rect.bottom / 2) - ((_Height + 39) / 2);
+
+		wchar_t* titleStr = title.Unicode();
+		Defer(delete[] titleStr);
+
+		// Create the window.
+		win32->hWnd = CreateWindowExW(
+			exStyle,
+			CLASS_NAME,
+			titleStr,
+			style,
+			rect.left,
+			rect.top,
+			_Width + 16, // Windows is weird, this makes it so that _Width and _Height
+			_Height + 39, // are the actually size of the window's viewport.
+			0,
+			0,
+			win32->hInstance,
+			0
+		);
+
+		if (win32->hWnd == NULL) {
+			ASSERT(0, "Win32: Failed to create a window.");
+			return;
+		}
+
+		SetWindowLongPtrW(win32->hWnd, GWLP_USERDATA, (LONG_PTR)this);
+
+		ShowWindow(win32->hWnd, SW_SHOW);
+		SetFocus(win32->hWnd);
+
+		// This is how you initialize modern OpenGL because Windows...
+		{
+			win32->hDC = GetDC(win32->hWnd);
+
+			PIXELFORMATDESCRIPTOR pfd = { sizeof(PIXELFORMATDESCRIPTOR) };
+			pfd.nVersion = 1;
+			pfd.dwFlags = PFD_DRAW_TO_WINDOW | PFD_SUPPORT_OPENGL | PFD_DOUBLEBUFFER;
+			pfd.iPixelType = PFD_TYPE_RGBA;
+			pfd.cColorBits = 32;
+			pfd.cDepthBits = 24;
+			pfd.cStencilBits = 8;
+			pfd.cAuxBuffers = 0;
+			pfd.iLayerType = PFD_MAIN_PLANE;
+			int format = ChoosePixelFormat(win32->hDC, &pfd);
+			SetPixelFormat(win32->hDC, format, &pfd);
+
+			InitGLExt();
+
+			int pixel_format_attribs[] = {
+				WGL_DRAW_TO_WINDOW_ARB,     GL_TRUE,
+				WGL_SUPPORT_OPENGL_ARB,     GL_TRUE,
+				WGL_DOUBLE_BUFFER_ARB,      GL_TRUE,
+				WGL_ACCELERATION_ARB,       WGL_FULL_ACCELERATION_ARB,
+				WGL_PIXEL_TYPE_ARB,         WGL_TYPE_RGBA_ARB,
+				WGL_COLOR_BITS_ARB,         32,
+				WGL_DEPTH_BITS_ARB,         24,
+				WGL_STENCIL_BITS_ARB,       8,
+				0
+			};
+
+			int pixel_format;
+			UINT num_formats;
+			wglChoosePixelFormatARB(win32->hDC, pixel_format_attribs, 0, 1, &pixel_format, &num_formats);
+			if (!num_formats) {
+				ASSERT(0, "Failed to set the OpenGL pixel format.");
+			}
+
+			//PIXELFORMATDESCRIPTOR pfd;
+			DescribePixelFormat(win32->hDC, pixel_format, sizeof(pfd), &pfd);
+			if (!SetPixelFormat(win32->hDC, pixel_format, &pfd)) {
```

Comment:
You stated above that the reason for the dummy window is because you can only set a pixel format once. This is the *second* call to `SetPixelFormat` on `win32->hDC`. (See L208) Is that intentional?